### DISCUSSION TITLE
Problem with SHRDZM Powermeter

### DISCUSSION THREAD

**Discussion by oszi-xx** [hoylabs/OpenDTU-OnBattery#1049] [Q&A]
I'm using a SHRDZM (ESP8266) to read my power meter data. Up to now it was working. After update to latest version
I got error if I'm using same settings (newer than 2024.01.04 )
[HttpPowerMeter] not a float: '69'
How can I  solve the problem .

**Comment by schlimmchen** [comment]
> [HttpPowerMeter] not a float: '69'

Is that an example or is that literally what it says on the serial/web console? `69` is something that can be parsed as a float, of course.

So you are now running 2024.06.03?

I can't reproduce this with an integer. I thought that maybe because the value is an integer, `value.is<float>()` could be false. That's not the case.

This issue seems to be reproducible by using a string: `{ "total": "123" }` gives me `[HttpPowerMeter] not a float: '123'`.

Could you please access the URL you configured as the power meter HTTP URL using a browser and share the result (the whole JSON) **verbatim** to confirm my suspicion?

**Comment by oszi-xx** [comment]
No "69" is just the actually power data. Yes I'm on 2024.06.03
{
"timestamp":"2024-06-16T07:52:40",
"32.7.0":"238.10",
"52.7.0":"238.70",
"72.7.0":"238.40",
"31.7.0":"0.00",
"51.7.0":"0.00",
"71.7.0":"1.36",
**"1.7.0":"223",**
"2.7.0":"0",
"1.8.0":"2877154",
"2.8.0":"41250",
"3.8.0":"51536",
"4.8.0":"499860",
**"16.7.0":"223",**
"uptime":"0000:20:07:15"
}

Console output
4D 00 7A 01 96 00 09 2F | -80 dBm
07:56:00.474 > Interrupt received
07:56:00.524 > RX Channel: 61 --> 95 80 11 37 42 80 11 37 42 02 1A 46 00 09 60 F5 00 00 00 28 09 4C 13 86 01 83 2D | -80 dBm
07:56:00.584 > MqttBattery: Updated voltage to 46.45 from 'Batteriespannung'
07:56:00.631 > Interrupt received
07:56:00.696 > RX Channel: 61 --> 95 80 11 37 42 80 11 37 42 83 00 00 00 10 03 E8 00 BD 00 03 A8 91 6A | -80 dBm
07:56:00.797 > RX Period End
07:56:00.848 > Success
07:56:00.916 > MqttBattery: Updated SoC to 9 from 'Batterie-SoC'
07:56:00.997 > MqttBattery: Updated voltage to 46.45 from 'Batteriespannung'
07:56:02.806 > MqttBattery: Updated SoC to 9 from 'Batterie-SoC'
07:56:02.864 > MqttBattery: Updated voltage to 46.45 from 'Batteriespannung'
07:56:02.922 > [HttpPowerMeter] Getting the power of phase 1 failed.
07:56:02.985 > [HttpPowerMeter] not a float: '210'
07:56:03.049 > [HttpPowerMeter] Getting the power of phase 1 failed.
07:56:03.153 > [HttpPowerMeter] not a float: '210'
07:56:03.202 > PowerMeterClass: TotalPower:  0.00**
07:56:03.252 > [HttpPowerMeter] Getting the power of phase 1 failed.
07:56:03.311 > [HttpPowerMeter] not a float: '210'
07:56:03.372 > MqttBattery: Updated SoC to 9 from 'Batterie-SoC'
07:56:03.460 > MqttBattery: Updated voltage to 46.45 from 'Batteriespannung'

**Reply by schlimmchen** [reply]
Thanks, this confirms my suspicion. This is quite annoying. This should be fixed in that powermeter reader, of course. JSON knows numbers for a reason. However, we shall not leave you hanging, it's a simple fix.

Are you ready for the next release, which requires an EPS32 with 8MB of flash? See issue 1025.

**Comment by oszi-xx** [comment]
Thanks for fast answer
Looks not good for me. I've only several 4MB types (ESP-32 Dev Kit C V4).
At the moment I'm going wit Mqtt and it works fine even its not working as standalone.
As i'm having many of them for testing and playing I hope there will be a fix for the old HW
If not I'll keep on going at the moment with Mqtt solution and latest SW.

**Comment by martind40** [comment]
Get the same issue at SHRDZM powermeter dongle. The SHRDZM Dongle send only integer values but OpenDTUoB expect a float value.
Is there a possibility to change something in the code that OpenDTUoB accept integer values to and show this in the dashboard?
After peressing "Test" at the "Stromzähler Einstellungen" slide I get the following massage (152 are the current power consumption, so OpenDTUoB read the correct value):
[HttpPowerMeter] not a float: '152'

**Reply by spcqike** [reply]
Im not to sure but maybe it was fixed with #1077? Maybe give it a try.

**Reply by martind40** [reply]
Thx. Tried it, but after flashing with ESPtool, I don't get any connection by WI-FI. Neither find the OpenDTU WiFI.

**Reply by martind40** [reply]
OK, the 8MB version is the wrong one for my ESP32. Now did it with the 4MB (no OTA) Version and set up alll parameters.
But still get an error after pressing "HTTP(S)-Anfrage(n) senden und Antworte(n) verarbeiten".
Error:
"Value 1: Value '58' at JSON path '1.7.0' is not of the expected type"

**Reply by martind40** [reply]
Ps: but the Value "58", was the correct value at that time!

**Reply by spcqike** [reply]
Well, thanks for your feedback. I guess @schlimmchen hasn‘t fixed this in the powermeter refactoring than (: 

can you use another way of submitting your data? Like MQTT? Until there is a fix for, I guess string type values?

**Reply by martind40** [reply]
It's not that big issue for me as I'm using HA as broker and there everything works for me.
But as there is the possibility to use the Smartmeter values directly in OpenDTUoB, I would make it testing and working.
If I can support developing the SW by giving some feedback, it's fine to me.

**Reply by spcqike** [reply]
Just out of curiosity, could you test MQTT, too, and tell, whether or not the module sends the power meter value as string there also? 

And, I case you don’t receive a message whenever you are mentioned somewhere else, please have a look at 
https://github.com/helgeerbe/OpenDTU-OnBattery/pull/1105

**Reply by martind40** [reply]
This is the send info if I subscribe the topic by using MQTTExplorer:
[b]sensor = { "timestamp":"2024-07-14T09:29:43", "1.8.0":"1002696", "1.8.1":"785056", "1.8.2":"217640", "1.7.0":"149", "2.8.0":"103340", "2.8.1":"103247", "2.8.2":"93", "2.7.0":"104", "3.8.0":"182904", "3.8.1":"134800", "3.8.2":"48104", "3.7.0":"0", "4.8.0":"123166", "4.8.1":"95436", "4.8.2":"27730", "4.7.0":"105", "16.7.0":"45", "uptime":"0000:12:02:00", "UTC":"2024-07-14T07:29:58" }[/b]
Also MQTTExplorer translate the string in a better readable table.
![Screenshot 2024-07-14 093353](https://github.com/user-attachments/assets/88df349a-4377-44e9-8a09-5e6082edba33)

**Reply by martind40** [reply]
I will try your mentioned #1105 

Edit: works! :-) see https://github.com/helgeerbe/OpenDTU-OnBattery/pull/1105#issuecomment-2227234590

**Reply by spcqike** [reply]
So the module sends strings everywhere. To bad. But good to know. Ty
